colossus/src/main/scala/colossus/protocols/http/Header.scala

```diff
@@ -197,7 +197,7 @@ class ParsedHttpHeaders(
     connection: Option[Connection]
   ) = this(headers.headers, transferEncodingOpt, contentLength, connection)
 
-  override def transferEncoding = transferEncodingOpt.getOrElse(TransferEncoding.Identity)
+  override def transferEncoding = if (transferEncodingOpt.isDefined) transferEncodingOpt.get else TransferEncoding.Identity
```

Comment:
assuming this is for performance reasons, right?  the first version is a bit more readable imho but, I'm cool with it if the benchmarks improve with it